### PR TITLE
[Feat] Per second burst rate limiting

### PR DESCRIPTION
**Pull request by justinkaseman**

### CODE REVIEW

File: packages/core/bootstrap/src/lib/burst-limit/index.ts [hunk 1]
```diff
@@ -13,37 +13,77 @@ import { AdapterError, logger } from '../external-adapter'
...
+
+    const observedRequestsInSecond = selectTotalNumberOfRequestsFor(requests, IntervalNames.SECOND)
+
+    if (observedRequestsInSecond > burstCapacity1s) {
+      logger.debug(
+        `Per Second Burst rate limit cap of ${burstCapacity1s} reached. ${observedRequestsInSecond} requests sent in the last minute. Waiting 1 second. Retry number: ${retry}`,
+      )
+      await delay(1000)
+    }
+    return true
```
Comment (cds95) [inline packages/core/bootstrap/src/lib/burst-limit/index.ts:37]: Should this be in the `if` block?

File: packages/core/bootstrap/src/lib/rate-limit/config.ts [hunk 2]
```diff
@@ -30,19 +31,27 @@ export function get(context: AdapterContext): Config {
...
   if (enabled) {
     const provider = getEnv('RATE_LIMIT_API_PROVIDER') || context.name?.toLowerCase() || ''
     const tier = getEnv('RATE_LIMIT_API_TIER') || ''
     try {
-      const providerConfig = getBurstLimit(provider, tier)
-      burstCapacity = Number(providerConfig)
+      const limit = getHTTPLimit(provider, tier, 'rateLimit1s')
+      burstCapacity1s = Number(limit)
+    } catch {
+      // Ignore
```
Comment (cds95) [inline packages/core/bootstrap/src/lib/rate-limit/config.ts:43]: Should we at least add some logs here for debugging?

**Review by cds95 — CHANGES_REQUESTED** [review]

File: packages/core/bootstrap/src/lib/burst-limit/index.ts (hunk 1 above)
Comment (justinkaseman) [reply packages/core/bootstrap/src/lib/burst-limit/index.ts:37]: No. This function is checking if we are above the limit on requests we sent in the last second.
If so (if block) delay a second and check again.
If not quit out by returning true.

File: packages/core/bootstrap/src/lib/rate-limit/config.ts (hunk 2 above)
Comment (justinkaseman) [reply packages/core/bootstrap/src/lib/rate-limit/config.ts:43]: The functions in their give their own more granular logs.

File: packages/core/bootstrap/src/lib/burst-limit/index.ts (hunk 1 above)
Comment (cds95) [reply packages/core/bootstrap/src/lib/burst-limit/index.ts:37]: but then won't this mean the `for` loop never goes past the first iteration as we'll always return `true` in the first iteration?

Comment (justinkaseman) [reply packages/core/bootstrap/src/lib/burst-limit/index.ts]: Ah shoot, you are right. Missing an `else` there.

**Review by cds95 — APPROVED** [review]

**Pull request merged**